sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/BinaryComparisonSimplificationSuite.scala

```diff
@@ -228,4 +228,31 @@ class BinaryComparisonSimplificationSuite extends PlanTest {
 
     checkCondition(nullableRelation, $"a" <=> nullLit, $"a".isNull)
   }
+
+  test("IN subquery nullability") {
+    // Should not optimize, since the IN subquery is nullable
+    {
```

Comment:
Please don't use this verbose redundant `{}` pattern.

You can use `Seq(...).foreach` patterns.